### ISSUE TITLE
Wait for Rethink's #4629 to get done

### ISSUE THREAD

**Issue by Slava** [Slava/meteor-rethinkdb#24]

**Comment by Slava** [comment]
https://github.com/rethinkdb/rethinkdb/issues/4629

**Comment by frankleng** [comment]
+1 to get updates

**Comment by muloka** [comment]
:+1: ditto as @frankleng

**Comment by faceyspacey** [comment]
+1

**Comment by faceyspacey** [comment]
https://github.com/rethinkdb/rethinkdb/issues/3579 , which was related to #4629, was done recently (7 days ago), not sure what bearing that has on #4629. RethinkDB seems like it would be HUGE for Meteor. @slava is this the only thing you need to complete the integration?

**Comment by jmusacchio** [comment]
+1

**Comment by Slava** [comment]
I don't think 3579 and 4629 were related. They are separate issues. I need both to be released in an official release to get the latency compensation working without race conditions.

**Comment by muloka** [comment]
Sounds very promising considering these are actively being worked on.

**Comment by dkran** [comment]
@muloka well, I believe this was all slated for 2.2 since they've been working on 2.1 so it's all sort of on schedule.

**Comment by muloka** [comment]
:+1:

**Comment by tabruhn** [comment]
Rethinkdb 2.2 is out today .. does the release solve the issue with latency compensation race conditions?

**Comment by Slava** [comment]
No, https://github.com/rethinkdb/rethinkdb/issues/4629 is still open.

**Comment by lalomartins** [comment]
Is this the only reason development is stalled? While this seems nice, I don't think it's relevant to most use cases… especially if you're creating ids on the client.

**Comment by Slava** [comment]
This is important if you want your Optimistic UI features to correctly roll-back if the server disagrees with the client's simulated changes. If you don't require that feature, I think the package should still work fine for you.

The development is stalled because it was a project for fun and I gave it a week to accomplish. And the result exceeded my expectations but the interest from the Meteor or RethinkDB community was never substantial enough for me to continue poring hours into it.
